docs/docs/rendering-elements.md

```diff
@@ -3,6 +3,79 @@ id: rendering-elements
 title: Rendering Elements
 permalink: docs/rendering-elements.html
 prev: introducing-jsx.html
+next: components-and-props.html
 ---
 
-TODO
+Elements are the smallest building blocks of React apps.
+
+An element describes what you want to see on the screen:
+
+```js
+const element = <h1>Hello, world</h1>;
+```
+
+Unlike browser DOM elements, React elements are plain objects, and are cheap to create. React DOM takes care of updating the DOM to match the React elements.
+
+## Rendering an Element into the DOM
+
+Let's say there is a `<div>` somewhere in your HTML file:
+
+```html
+<div id="root"></div>
+```
+
+We call this a "container" DOM node because the React app is contained by it. Everything inside it will be managed by React DOM.
+
+Applications built with just React usually have a single container node. If you are integrating React into an existing app, you may have as many isolated container nodes as you like.
+
+To render a React element into a DOM container, pass both to `ReactDOM.render()`:
+
+```js{3}
+const container = document.getElementById('root');
+const element = <h1>Hello, world</h1>;
+ReactDOM.render(element, container);
+```
+
+[Try it on Codepen.](http://codepen.io/gaearon/pen/rrpgNB?editors=1010)
+
+It displays "Hello World" on the page.
+
+## Updating the Rendered Element
+
+React elements are [immutable](https://en.wikipedia.org/wiki/Immutable_object). Once you create an element, you can't change it.
+
+When we want the UI to update, we create a new element, and pass it to `ReactDOM.render()`. It updates the DOM to match the newly passed element.
```

Comment:
I added this note. I hope it helps.

I don't want to remove the example because people often don't realize how `setState()` is essentially the same thing as `ReactDOM.render()`, but encapsulated. This knowledge is valuable IMO, and the whole state article builds up on this specific example.

<img width="832" alt="screen shot 2016-10-07 at 16 10 22" src="https://cloud.githubusercontent.com/assets/810438/19195228/976000b8-8ca8-11e6-9029-a1774cecbfa4.png">